lib/utils/vpc-utils.ts

```diff
@@ -2,14 +2,37 @@ import * as ec2 from 'aws-cdk-lib/aws-ec2';
 import { Stack } from 'aws-cdk-lib';
 import { AwsCustomResource, AwsCustomResourcePolicy, AwsSdkCall } from "aws-cdk-lib/custom-resources";
 
+/**
+ * Tags EC2 Security Group with given tag and value - used for EKS Security Group Tagging
+ * @param stack - CDK Stack
+ * @param securityGroupId - Security Group Resource ID
+ * @param key - Tag Key
+ * @param value - Tag Value
+ */
+export function tagSecurityGroup(stack: Stack, securityGroupId: string, key: string, value: string): void {
+    const tags = [{
+        Key: key,
+        Value: value
+    }];
+
+    const arn = `arn:${stack.partition}:ec2:${stack.region}:${stack.account}:security-group/`+securityGroupId;
+
+    const parameters = {
+        Resources: [securityGroupId],
+        Tags: tags
+    };
+
+    applyEC2Tag("eks-sg", stack, parameters, key, [arn]);
+}
+
 /**
  * Tags VPC Subnets with given tag and value.
- * @param stack 
- * @param subnetArns 
- * @param tag 
- * @param value 
+ * @param stack - CDK Stack
+ * @param subnets - a list of subnets
+ * @param key - Tag Key
+ * @param value - Tag Value
  */
-export function tagSubnets(stack: Stack, subnets: ec2.ISubnet[], tag: string, value: string): void {
+export function tagSubnets(stack: Stack, subnets: ec2.ISubnet[], key: string, value: string): void {
```

Comment:
Is there an issue against CDK to support this tagging without custom resources? I recall you mentioned something related to it before. 
If there is no issue:
1. let's create an issue against CDK (customers are very reluctant to see lambda functions in their accounts)
2. let's create an issue against this repo to update this logic once CDK adds native support.

If there an issue already, then only 2/ applies.

Reply:
I think I raised an issue, let me confirm. It had to do with tokens not resolving for us to use CDK native Tags. I will create an issue against this repo as well. Consider it addressed.